charts/dvpe-deployment-gloo/templates/secret.yaml

```diff
@@ -2,58 +2,91 @@
 metadata:
   name: {{ $serviceName }}-service-secrets
   namespace: {{ .Release.Namespace }}
 spec:
-  backendType: secretsManager
-  {{- if .Values.externalSecrets.service.roleArn }}
-  roleArn: {{ .Values.externalSecrets.service.roleArn }}
-  {{- end}}
+  secretStoreRef:
+    kind: SecretStore
+  {{- if .Values.externalSecrets.service.secretstoreName }}
+    name: {{ .Values.externalSecrets.service.secretstoreName }}
+  {{- else }}
+    name: external-secret-store-{{ .Release.Namespace }}
+  {{- end }}
+  target:
+    name: {{ $serviceName }}-service-secrets
   dataFrom:
-    - {{ .Values.externalSecrets.service.key }}
+  - extract:
+      key: {{ .Values.externalSecrets.service.key }}
+  {{- if .Values.externalSecrets.service.refreshInterval }}
+  refreshInterval: {{ .Values.externalSecrets.service.refreshInterval }}
+  {{- else }}
+  refreshInterval: 15m
+  {{- end }}
 {{- end }}
 
 {{- if and .Values.externalSecrets.oauth2.key .Values.gloo.authConfig.spec.configs.oauth.clientId }}
 ---
-apiVersion: 'kubernetes-client.io/v1'
+apiVersion: 'external-secrets.io/v1beta1'
 kind: ExternalSecret
 metadata:
   name: {{ $serviceName }}-oidc-secrets
   namespace: {{ .Release.Namespace }}
 spec:
-  backendType: secretsManager
-  {{- if .Values.externalSecrets.service.roleArn }}
-  roleArn: {{ .Values.externalSecrets.service.roleArn }}
-  {{- end}}
-  template:
-    metadata:
-      annotations:
-        resource_kind: '*v1.Secret'
+  secretStoreRef:
+    kind: SecretStore
+  {{- if .Values.externalSecrets.service.secretstoreName }}
+    name: {{ .Values.externalSecrets.service.secretstoreName }}
+  {{- else }}
+    name: external-secret-store-{{ .Release.Namespace }}
+  {{- end }}
+  target:
+    name: {{ $serviceName }}-oidc-secrets
+    template:
+      metadata:
+        annotations:
+          resource_kind: '*v1.Secret'
   data:
-    - key: {{ .Values.externalSecrets.oauth2.key }}
-      property: {{ .Values.gloo.authConfig.spec.configs.oauth.clientId | quote }}
-      name: oauth
+    - secretKey: oauth
+      remoteRef:
+        key: {{ .Values.externalSecrets.oauth2.key }}
+        property: {{ .Values.gloo.authConfig.spec.configs.oauth.clientId | quote }}
+  {{- if .Values.externalSecrets.service.refreshInterval }}
+  refreshInterval: {{ .Values.externalSecrets.service.refreshInterval }}
+  {{- else }}
+  refreshInterval: 15m
+  {{- end }}
 
 {{- end }}
 
 {{- if and .Values.externalSecrets.oauth2.key .Values.gloo.authConfig.spec.configs.clientCredentialsPlugin.config.clientId }}
 ---
-apiVersion: 'kubernetes-client.io/v1'
+apiVersion: 'external-secrets.io/v1beta1'
 kind: ExternalSecret
 metadata:
   name: {{ $serviceName }}-oauth2-client-credentials-secrets
   namespace: {{ .Release.Namespace }}
 spec:
-  backendType: secretsManager
-  {{- if .Values.externalSecrets.service.roleArn }}
-  roleArn: {{ .Values.externalSecrets.service.roleArn }}
+  secretStoreRef:
+    kind: SecretStore
+  {{- if .Values.externalSecrets.service.secretstoreName }}
```

Comment:
We could use the "default" function here.
Don't know what is best practice.
https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function

Reply:
Same goes for the other occurrences of this if-else-statement ofc

Reply:
Code has been changed and tested.